test/e2e/specs/editor/various/inner-blocks-templates.spec.js

```diff
@@ -28,9 +28,11 @@ test.describe( 'Inner blocks templates', () => {
 			name: 'test/test-inner-blocks-async-template',
 		} );
 
-		const blockWithTemplateContent = editor.canvas.locator(
-			'role=document[name="Block: Test Inner Blocks Async Template"i] >> text=OneTwo'
-		);
+		const blockWithTemplateContent = page
+			.frameLocator( '[name=editor-canvas]' )
```

Comment:
Could you explain why we need to use `frameLocator` here? Is it still needed if we wait for the page to load in `createNewPost`?

Reply:
It's for reuse after the `page.reload()` that follows. Now that this test runs in the iframed editor `editor.canvas` returns a [`Frame`](https://playwright.dev/docs/api/class-frame) and the locator formed from that isn't reusable after a page reload.

So this can be done differently if there's a style that seems better to you or I can add a comment. I thought it might be ideal to consider doing a follow to try having `editor.canvas` return a `FrameLocator` instead of a `Frame` and then this could revert back to using that.

Reply:
> I thought it might be ideal to consider doing a follow up to try having editor.canvas return a FrameLocator instead of a Frame and then this could revert back to using that.

Yeah, I regret using `frame` in the first place 😅 . One thing to consider is that there are code that depend on some `frame`-specific API so it's not easy to switch to `frameLocator` now. I've been trying to find alternatives but haven't found any. Please share if you have an idea! 🙇

Reply:
> One thing to consider is that there are code that depend on some frame-specific API so it's not easy to switch to frameLocator now

I'd noticed but I figured it should possible to refactor any dependent tests/code. It also looks like many of the `Frame` methods are deprecated so it might be good for scrubbing that in case we are using some.  I don't know Playwright well enough to have a clue if there's any easier alternative.

Reply:
I couldn't find an alternative to `evaluate` or `waitForFunction` though, which I believe some existing code rely on?

Reply:
Thanks for those specifics. It does look like that is a sticking point 🤔.